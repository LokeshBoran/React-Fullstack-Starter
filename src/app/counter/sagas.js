import { delay } from 'redux-saga';
import { select, put, takeEvery, all } from 'redux-saga/effects';

import { RESET, RESET_ASYNC, RESET_IF_ODD, RESET_IF_EVEN } from './constants';

export function *resetAsync() {
  yield delay(1000);
  yield put({ type: RESET });
}

export function *watchResetAsync() {
  yield takeEvery(RESET_ASYNC, resetAsync);
}

export function *resetIfOdd() {
  const { value } = yield select(({ counter }) => counter);

  if (Math.abs(value % 2) === 1) {
    yield put({ type: RESET });
  }
}

export function *watchResetIfOdd() {
  yield takeEvery(RESET_IF_ODD, resetIfOdd);
}

export function *resetIfEven() {
  const { value } = yield select(({ counter }) => counter);

  if (value % 2 === 0) {
    yield put({ type: RESET });
  }
}

export function *watchResetIfEven() {
  yield takeEvery(RESET_IF_EVEN, resetIfEven);
}

export default function *() {
  yield all([
    watchResetAsync(),
    watchResetIfOdd(),
    watchResetIfEven()
  ]);
}
