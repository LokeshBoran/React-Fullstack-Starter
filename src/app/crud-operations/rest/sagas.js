import { put, takeEvery, all } from 'redux-saga/effects';
import axios from 'axios';

import { API_LIST, ADD_ITEM_SAGA, SEARCH_ITEM_SAGA, EDIT_ITEM_SAGA, DELETE_ITEM_SAGA } from './constants';
import { success, failure, setData, searchItemSaga } from './actions';

export function *effectAddItemSaga({ text }) {
  try {
    yield axios.post(API_LIST, { text });
    yield put(searchItemSaga());
  } catch (error) {
    put(failure(error));
  }
}

export function *watchAddItemSaga() {
  yield takeEvery(ADD_ITEM_SAGA, effectAddItemSaga);
}

export function *effectSearchItemSaga({ text }) {
  try {
    const { data } = yield axios.get(API_LIST, { params: { text } });
    yield put(success(data));
    yield put(setData({ loading: false }));
  } catch (error) {
    put(failure(error));
  }
}

export function *watchSearchItemSaga() {
  yield takeEvery(SEARCH_ITEM_SAGA, effectSearchItemSaga);
}

export function *effectEditItemSaga({ id, text }) {
  try {
    yield axios.put(`${API_LIST}/${id}`, { text });
    yield put(searchItemSaga());
  } catch (error) {
    put(failure(error));
  }
}

export function *watchEditItemSaga() {
  yield takeEvery(EDIT_ITEM_SAGA, effectEditItemSaga);
}

export function *effectDeleteItemSaga({ id }) {
  try {
    yield axios.delete(`${API_LIST}/${id}`);
    yield put(searchItemSaga());
  } catch (error) {
    put(failure(error));
  }
}

export function *watchDeleteItemSaga() {
  yield takeEvery(DELETE_ITEM_SAGA, effectDeleteItemSaga);
}

export default function *() {
  yield all([
    watchAddItemSaga(),
    watchSearchItemSaga(),
    watchEditItemSaga(),
    watchDeleteItemSaga()
  ]);
}
