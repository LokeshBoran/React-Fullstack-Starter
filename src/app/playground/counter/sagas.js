import { delay } from 'redux-saga';
import { select, put, takeLatest, all } from 'redux-saga/effects';

import { RESET_ASYNC, RESET_IF_EVEN, RESET_IF_ODD } from './constants';
import { reset } from './actions';

export function *resetAsync() {
  yield delay(1000);
  yield put(reset());
}

export function *watchResetAsync() {
  yield takeLatest(RESET_ASYNC, resetAsync);
}

export function *resetIfEven() {
  const { value } = yield select(({ counter }) => counter);
  if (value % 2 === 0) yield put(reset());
}

export function *watchResetIfEven() {
  yield takeLatest(RESET_IF_EVEN, resetIfEven);
}

export function *resetIfOdd() {
  const { value } = yield select(({ counter }) => counter);
  if (Math.abs(value % 2) === 1) yield put(reset());
}

export function *watchResetIfOdd() {
  yield takeLatest(RESET_IF_ODD, resetIfOdd);
}

export default function *() {
  yield all([
    watchResetAsync(),
    watchResetIfEven(),
    watchResetIfOdd()
  ]);
}
