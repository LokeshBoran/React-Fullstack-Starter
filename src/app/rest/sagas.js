import { put, takeEvery, all } from 'redux-saga/effects';
import axios from 'axios';

import { API_LIST, ADD_ITEM_SAGA, SEARCH_ITEM_SAGA, EDIT_ITEM_SAGA, DELETE_ITEM_SAGA } from './constants';
import { success, searchItemSaga, setData } from './actions';

export function *effectAddItemSaga({ text }) {
  yield axios.post(API_LIST, { text });
  yield put(searchItemSaga());
}

export function *watchAddItemSaga() {
  yield takeEvery(ADD_ITEM_SAGA, effectAddItemSaga);
}

export function *effectSearchItemSaga({ text }) {
  const { data } = yield axios.get(text ? `${API_LIST}?text=${text}` : API_LIST);
  yield put(success(data));
  yield put(setData({ loading: false }));
}

export function *watchSearchItemSaga() {
  yield takeEvery(SEARCH_ITEM_SAGA, effectSearchItemSaga);
}

export function *effectEditItemSaga({ id, text }) {
  yield axios.put(`${API_LIST}/${id}`, { text });
  yield put(searchItemSaga());
}

export function *watchEditItemSaga() {
  yield takeEvery(EDIT_ITEM_SAGA, effectEditItemSaga);
}

export function *effectDeleteItemSaga({ id }) {
  yield axios.delete(`${API_LIST}/${id}`);
  yield put(searchItemSaga());
}

export function *watchDeleteItemSaga() {
  yield takeEvery(DELETE_ITEM_SAGA, effectDeleteItemSaga);
}

export default function *() {
  yield all([
    watchAddItemSaga(),
    watchSearchItemSaga(),
    watchEditItemSaga(),
    watchDeleteItemSaga()
  ]);
}
