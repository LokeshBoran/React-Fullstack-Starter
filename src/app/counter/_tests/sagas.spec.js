import { resetAsync } from '../sagas';

describe('Counter', () => {
  describe('sagas', () => {
    it('should call get reset', () => {
      const generator = resetAsync();
      generator.next();
      const putResult = generator.next();
      expect(putResult).toMatchSnapshot();
    });
  });
});
