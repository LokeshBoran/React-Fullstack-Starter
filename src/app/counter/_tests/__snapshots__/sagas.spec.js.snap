// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter sagas should call get reset 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "type": "[Counter] RESET",
      },
      "channel": null,
    },
  },
}
`;
